pkg/controller/forwarding/forwarding_controller.go

```diff
@@ -0,0 +1,127 @@
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileForwarding{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("logforwarding-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ClusterLogging
+	err = c.Watch(&source.Kind{Type: &logforwarding.LogForwarding{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileForwarding{}
+
+// ReconcileForwarding reconciles a ClusterLogging object
+type ReconcileForwarding struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+var (
+	reconcilePeriod = 30 * time.Second
+	reconcileResult = reconcile.Result{RequeueAfter: reconcilePeriod}
+)
+
+// Reconcile reads that state of the cluster for a LogForwarding object and makes changes based on the state read
+// and what is in the LogForwarding.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileForwarding) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the LogForwarding instance
+	instance := &logforwarding.LogForwarding{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	//check for instancename and then update status
+	if instance.Name == singletonName {
+		instance.Status.LogForwardingStatus = logforwarding.LogForwardingStatus{
+			LogForwardingState: logforwarding.LogForwardingStateAccepted,
+		}
+	} else {
+		instance.Status.LogForwardingStatus = logforwarding.LogForwardingStatus{
+			LogForwardingState: logforwarding.LogForwardingStateRejected,
+			Reason:             singletonMessage,
+		}
+	}
+	if err = r.client.Status().Update(context.TODO(), instance); err != nil {
+		return reconcile.Result{}, err
+	}
+	if instance.Name != singletonName {
+		return reconcile.Result{}, nil
+	}
+
+	clInstance := &loggingv1.ClusterLogging{}
+	clName := types.NamespacedName{Name: singletonName, Namespace: openshiftNS}
```

Comment:
i think if we don't find the `clInstance` we shouldn't reconcile the same... 
should we split up the logic in `k8shandler.Reconcile` and move it into the controllers?

Reply:
> i think if we don't find the `clInstance` we shouldn't reconcile the same...
Why not? LF depends on CL to evaluate whether it should default routing to the operator managed logstore.  CL relies on LF to understand how to create the collector config.  Presumably there should be no issues processing the same since we aren't saving CL since its an empty struct.  If it is then we should fix that.
  
> should we split up the logic in `k8shandler.Reconcile` and move it into the controllers?
Not sure I understand the benefit

Reply:
> Why not? LF depends on CL to evaluate whether it should default routing to the operator managed logstore. CL relies on LF to understand how to create the collector config.

I'm referring to there being no CL... meaning our operator shouldn't be processing the CL at all (the current functionality). This controller is only going to fire from LF events. we would expect that our CL be stood up by the other controller

> Not sure I understand the benefit

Code readibility

Reply:
> > Why not? LF depends on CL to evaluate whether it should default routing to the operator managed logstore. CL relies on LF to understand how to create the collector config.
> 
> I'm referring to there being no CL... meaning our operator shouldn't be processing the CL at all (the current functionality). This controller is only going to fire from LF events. we would expect that our CL be stood up by the other controller

The state of LF directly affects the state of CL by way of the collector config.  You change pipelines and outputs it will be reflected in new fluent.conf and a bump in the fluent DS.  Assuming we have coded the rest of the CL logic properly there should be no reason for anything else about the stack to change; if it does then that is a bug.  I'm not sure I fully understand your concerns given the resources are intertwined and need to be reconciled as a pair.  The intention overall is to drive our operands to a desired state which can and will occur starting with either controller